builtInFunctions/esdtSetTokenType.go

```diff
@@ -0,0 +1,94 @@
+package builtInFunctions
+
+import (
+	"bytes"
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-core-go/marshal"
+	"github.com/multiversx/mx-chain-vm-common-go"
+)
+
+const (
+	tokenIDindex   = 0
+	tokenTypeIndex = 1
+)
+
+type esdtSetTokenType struct {
+	baseActiveHandler
+	globalSettingsHandler vmcommon.GlobalMetadataHandler
+	accounts              vmcommon.AccountsAdapter
+	marshaller            marshal.Marshalizer
+}
+
+// NewESDTSetTokenTypeFunc returns the esdt set token type built-in function component
+func NewESDTSetTokenTypeFunc(
+	accounts vmcommon.AccountsAdapter,
+	globalSettingsHandler vmcommon.GlobalMetadataHandler,
+	marshaller marshal.Marshalizer,
+	activeHandler func() bool,
+) (*esdtSetTokenType, error) {
+	if check.IfNil(accounts) {
+		return nil, ErrNilAccountsAdapter
+	}
+	if check.IfNil(marshaller) {
+		return nil, ErrNilMarshalizer
+	}
+	if check.IfNil(globalSettingsHandler) {
+		return nil, ErrNilGlobalSettingsHandler
+	}
+	if activeHandler == nil {
+		return nil, ErrNilActiveHandler
+	}
+
+	e := &esdtSetTokenType{
+		accounts:              accounts,
+		globalSettingsHandler: globalSettingsHandler,
+		marshaller:            marshaller,
+	}
+
+	e.baseActiveHandler.activeHandler = activeHandler
+
+	return e, nil
+}
+
+// ProcessBuiltinFunction saves the token type in the system account
+func (e *esdtSetTokenType) ProcessBuiltinFunction(_, _ vmcommon.UserAccountHandler, vmInput *vmcommon.ContractCallInput) (*vmcommon.VMOutput, error) {
+	if vmInput == nil {
```

Comment:
is this builtin function meant to be called directly from a transaction? If yes, where do we check the gas provided?

Reply:
This is meant to be called only from ESDTSystemSc. Correct me if I'm wrong @sasurobert.